src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs

```diff
@@ -194,7 +200,102 @@ public void SetupEnvironment(AzureModule mode)
 #endif
         }
 
-        private void SetupAzureEnvironmentFromEnvironmentVariables(AzureModule mode)
+        public void SetEnvironmentVariableFromCredentialFile()
+        {
+            var filePath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + "/.azure/testcredentials.json";
+            Dictionary<string, object> credentials;
+            using (StreamReader r = new StreamReader(filePath))
+            {
+                string json = r.ReadToEnd();
+                credentials = JsonUtilities.DeserializeJson(json);
+            }
+
+            if (Environment.GetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION") == null)
+            {
+                StringBuilder formattedConnectionString = new StringBuilder();
+                formattedConnectionString.Append(String.Format("SubscriptionId={0};HttpRecorderMode={1};Environment={2}", credentials["SubscriptionId"], credentials["HttpRecorderMode"], credentials["Environment"]));
```

Comment:
You canjust use StringBuilder.AppendFormat': https://msdn.microsoft.com/en-us/library/cazfhf32(v=vs.110).aspx